pennylane/tape/tapes/jacobian_tape.py

```diff
@@ -490,24 +490,28 @@ def jacobian(self, device, params=None, **options):
         all_tapes = []
         reshape_info = []
         processing_fns = []
+        nonzero_grad_idx = []
 
         for trainable_idx, param_method in enumerate(diff_methods):
+            if param_method != "0":
 
-            if (method == "best" and param_method[0] == "F") or (method == "numeric"):
-                # numeric method
-                tapes, processing_fn = self.numeric_pd(trainable_idx, params=params, **options)
+                nonzero_grad_idx.append(trainable_idx)
```

Comment:
Minor suggestion: you could do the following,
```python
for trainable_idx, param_method in enumerate(diff_methods):
    if param_method == "0":
        continue

    nonzero_grad_idx.append(trainable_idx)

    if (method == "best" and param_method[0] == "F") or (method == "numeric"):
        # numeric method
        tapes, processing_fn = self.numeric_pd(trainable_idx, params=params, **options)

    elif (method == "best" and param_method[0] == "A") or (method == "analytic"):
        # analytic method
        tapes, processing_fn = self.analytic_pd(trainable_idx, params=params, **options)

    processing_fns.append(processing_fn)

    # we create a flat list here to feed at once to the device
    all_tapes.extend(tapes)

    # to extract the correct result for this parameter later, remember the number of tapes
    reshape_info.append(len(tapes))
```

While no change to the logic, it helps with readability (since it avoids the extra indent).

Reply:
Thanks, done!